### PR TITLE
todoman: 4.4.0 -> 4.5.0

### PR DESCRIPTION
**Pull request by antonmosich**
See https://github.com/pimutils/todoman/blob/v4.5.0/CHANGELOG.rst#L7-L18 for the Changelog of v4.5.0. I also ran `nixfmt` and added `sphinxHook` so that the docs are also built. For the latter it was necessary to package `sphinx-click`, a sphinx plugin for CLI applications using `click`.
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.sphinx-click</li>
    <li>python311Packages.sphinx-click.dist</li>
    <li>python312Packages.sphinx-click</li>
    <li>python312Packages.sphinx-click.dist</li>
    <li>todoman</li>
    <li>todoman.dist</li>
  </ul>
</details>

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/sphinx-click/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  sphinx,
+}:
+
+buildPythonPackage rec {
+  pname = "sphinx-click";
+  version = "6.0.0";
+  pyproject = true;
+
+  build-system = [
+    setuptools-scm
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinx-click/default.nix:21]:
```suggestion
    setuptools
```

File: pkgs/development/python-modules/sphinx-click/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  nativeBuildInputs = [
+    sphinxHook
+  ];
+
+  dependencies = [
+    click
+    docutils
+    pbr
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinx-click/default.nix:31]: `pbr` should go to `build-system`.
https://github.com/click-contrib/sphinx-click/blob/7f29172b10d4dc251af414ae01e38c3b970686a1/setup.py#L6

File: pkgs/development/python-modules/sphinx-click/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    sphinx
+  ];
+
+  src = fetchPypi {
+    inherit version;
+    pname = "sphinx_click";
+    hash = "sha256-9dZkMh3AxmIv8Bnx4chOWM4M7P3etRDgBM9gwqOrRls=";
+  };
+
```
Comment (natsukium) [inline pkgs/development/python-modules/sphinx-click/default.nix:40]: Could you enable tests with `pytestCheckHook` and `pythonImportsCheck`?

**Comment by antonmosich** [comment]
Thanks for your review, I think I addressed all your comments.

File: pkgs/applications/office/todoman/default.nix
Comment (FliegendeWurst) [inline pkgs/applications/office/todoman/default.nix:1]: Build fails
```
Executing pytestCheckPhase
ImportError while loading conftest '/build/source/tests/conftest.py'.
tests/conftest.py:14: in <module>
    import pytz
E   ModuleNotFoundError: No module named 'pytz'
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347956`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python311Packages.sphinx-click</li>
    <li>python312Packages.sphinx-click</li>
    <li>todoman</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python311Packages.sphinx-click</li>
    <li>python312Packages.sphinx-click</li>
    <li>todoman</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python311Packages.sphinx-click</li>
    <li>python312Packages.sphinx-click</li>
    <li>todoman</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python311Packages.sphinx-click</li>
    <li>python312Packages.sphinx-click</li>
    <li>todoman</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**